pydantic/schema.py

```diff
@@ -148,6 +148,31 @@ def model_schema(
     return m_schema
 
 
+def get_field_info_schema(field: ModelField) -> Tuple[Dict[str, Any], bool]:
+    schema_overrides = False
+
+    # If no title is explicitly set, we don't set title in the schema for enums.
+    # The behaviour is the same as `BaseModel` reference, where the default title
+    # is in the definitions part of the schema.
+    is_enum_with_no_title_set = lenient_issubclass(field.type_, Enum) and not field.field_info.title
+    schema: Dict[str, Any] = {} if is_enum_with_no_title_set else {
+        'title': field.field_info.title or field.alias.title().replace('_', ' ')
+    }
```

Comment:
```suggestion
    schema: Dict[str, Any] = {}
    if lenient_issubclass(field.type_, Enum) and not field.field_info.title:
        schema['title'] = field.field_info.title or field.alias.title().replace('_', ' ')
```
or something would be clearer.